### ISSUE TITLE
SlideFlow-Studio shows wrong percentages of probability of the feature  

### ISSUE THREAD

**Issue by frasca17** [slideflow/slideflow#405]
I trained a model to predict the presence/absence of a mutation in a gene on H&E slides. 

When I want to exploit SlideFlow Studio in order to visualize the prediction live, once I load the Project and the model, if I select a slide on which I want to "right click" in order to see the live predictions, it results in probability percentages higher than 100% or lower than 0. 

For instance, if I right click on a tile to see the prediction percentage, I can get 478% of my tile to be mutated or -972% to not be mutated.

Moreover, if I click on the white space surrounding the image, I get again the probability of the prediction, also if no tissue is present.

How is it all this possible?

What am I doing wrong to get these weired numbers?

Thank you very much

**Comment by jamesdolezal** [comment]
One additional comment - if you generate a whole-slide prediction, the displayed heatmap should reflect post-softmax values. You can see these values by holding the <ALT> key on your keyboard and hovering over the slide. The heatmap will also automatically skip the whitespace areas.

Thanks again for raising the issue!

**Comment by frasca17** [comment]
Dear James,

Thank you very much for your reply and for the tips regarding the heatmaps!

I guess this could certainly help me to visualize the training results!


I was actually wondering if uncertainty estimates should be always enabled during training, since I would like to give a statistical meaning to the predictions. I know that focal predictions are not the best, and thank you very much for the clarifications.

Another question that I have is regarding the possibility to measure (in cm^2 or um) the surface of the predicted area. Is it possible through SlideFlow Studio? I am a bioinformatician, but my pathologitst colleagues are always asking me if I am able to quantify the predicted area.


Thank you again for availability and for the amazing software you created!

**Comment by jamesdolezal** [comment]
Fortunately, UQ can actually be enabled *after* training as well. Just edit the `params.json` file in the model folder, changing the following line:
```json
"uq": false
```
to
```json
"uq": true
```
Then, any time the model is loaded, UQ will be enabled. UQ estimates are made via ensemble, calculating standard deviation of model predictions with 30 forward passes during dropout-enabled inference. [Our paper](https://www.nature.com/articles/s41467-022-34025-x) describing this algorithm includes additional statistical information (see the Methods section).

We recommend enabling UQ for all final tile-based models after you're done with initial hyperparamater selection. The only downside is that it increases the time required to generate a prediction by 30-fold. We have an optimization coming in version 3.0 that eliminates this, but for now, be aware that it will take longer to generate predictions using UQ.

Regarding the question about measuring surface area - we do have some preliminary function that allows for quantifying tumor area for segmentation models, but not yet for quantifying area based on tile-based model predictions. The interface that exists is:

```python

import slideflow as sf
from slideflow.slide import qc

segment = qc.Segment('.../path_to_segmentation_model')
wsi = sf.WSI('slide.svs', tile_px=256, tile_um=256, ...)
wsi.qc(segment)
print("Area of segmented tumor (mm^2): ", wsi.area())
```

Adding some additional tools to do similar analyses with tile-based models is a great suggestion - we'll add it to the list.

**Comment by jamesdolezal** [comment]
Fixed in version 3.0

**Comment by frasca17** [comment]
Thank you very much for the help!
